integration-tests/src/query_params.test.js

```diff
@@ -93,6 +93,65 @@ describe("for a GET endpoint with no parameters", () => {
   });
 });
 
+const testOptionalEndpoint: SafeAPI.Endpoint<
+  {
+    x: ?string,
+    y: string
+  },
+  {
+    x: ?string,
+    y: string
+  }
+> = SafeAPI.endpoint()
+  .fragment("foo")
+  .queryParams({
+    x: T.maybeString,
+    y: T.string
+  });
+
+const testOptionalHandler = async input => {
+  return {
+    x: input.x ? input.x.toString() : null,
+    y: input.y
+  };
+};
+
+describe("for a GET endpoint with optional query parameters", () => {
+  let server;
+  beforeEach(() => {
+    server = TestUtils.makeServer({
+      endpoint: testOptionalEndpoint,
+      handler: testOptionalHandler
+    });
+  });
+
+  test("server can parse the input when present", async () => {
+    const resp = await fetch(
+      `http://localhost:${server.address().port}/foo?x=X&y=Y`
+    );
+    const json = await resp.json();
+    expect(json).toEqual({ x: "X", y: "Y" });
+  });
+
+  test("server will parse null inputs", async () => {
+    const resp = await fetch(
+      `http://localhost:${server.address().port}/foo?x=&y=Y`
+    );
+    const json = await resp.json();
+    expect(json).toEqual({ x: null, y: "Y" });
+  });
+
+  test("server will accept absent optional inputs", async () => {
+    const resp = await fetch(
+      `http://localhost:${server.address().port}/foo?y=Y`
+    );
+    const json = await resp.json();
+
+    expect(json).toEqual({ x: null, y: "Y" });
+    expect(Object.keys(json)).toEqual(["x", "y"]);
```

Comment:
redundant assertion

Reply:
So... This is to explicitly test that the object contains `x` as a key with a null value. The expression `toEqual({ x: null, y: "Y" })` probably throws an error on `{y: "Y"}`, but I think this more clear of what's being tested from a doc point of view.

Reply:
Check to see whether `.toEqual({x: null, y: "Y"})` works with an object that does not contain `x` (I would guess it doesn't). If it doesn't I think the test above should be clear enough!

Reply:
never mind. This test is already written above.